File: app/src/main/java/com/nanchen/aiyagirl/module/category/CategoryContract.java
package com.nanchen.aiyagirl.module.category;

import com.nanchen.aiyagirl.base.BasePresenter;
import com.nanchen.aiyagirl.base.BaseView;
import com.nanchen.aiyagirl.model.CategoryResult.ResultsBean;

import java.util.List;

/**
 * Author: nanchen
 * Email: liushilin520@foxmail.com
 * Date: 2017-04-14  10:14
 */

public interface CategoryContract {

    interface ICategoryView extends BaseView{
        
        void getCategoryItemsFail(String failMessage);

        void setCategoryItems(List<ResultsBean> data);

        void addCategoryItems(List<ResultsBean> data);

        void showSwipeLoading();

        void hideSwipeLoading();

        void setLoading();

        String getCategoryName();

        void setNoMore();
    }
    
    interface ICategoryPresenter extends BasePresenter{
        
        void getCategoryItems(boolean isRefresh);
    }
}

File: app/src/main/java/com/nanchen/aiyagirl/module/category/CategoryContract.kt
package com.nanchen.aiyagirl.module.category

import com.nanchen.aiyagirl.base.BasePresenter
import com.nanchen.aiyagirl.base.BaseView
import com.nanchen.aiyagirl.model.ResultsBean

/**
 * Author: nanchen
 * Email: liushilin520@foxmail.com
 * Date: 2017-04-14  10:14
 */

interface CategoryContract {

    interface ICategoryView : BaseView {

        val categoryName: String

        fun getCategoryItemsFail(failMessage: String)

        fun setCategoryItems(data: List<ResultsBean>)

        fun addCategoryItems(data: List<ResultsBean>)

        fun showSwipeLoading()

        fun hideSwipeLoading()

        fun setLoading()

        fun setNoMore()
    }

    interface ICategoryPresenter : BasePresenter {

        fun getCategoryItems(isRefresh: Boolean)
    }
}
